### PR TITLE
add appvendor managerment

### PR DESCRIPTION
**Pull request by huojiao2006**

### CODE REVIEW

File: pkg/models/vendor_verify_info.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// that can be found in the LICENSE file.
+
+package models
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/logger"
+
```
Comment (chilianyi) [inline pkg/models/vendor_verify_info.go]: Useless  blank line

File: pkg/models/vendor_verify_info.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	Vendor.AuthorizerEmail = req.GetAuthorizerEmail().GetValue()
+	Vendor.AuthorizerPhone = req.GetAuthorizerPhone().GetValue()
+	Vendor.BankName = req.GetBankName().GetValue()
+	Vendor.BankAccountName = req.GetBankAccountName().GetValue()
+	Vendor.BankAccountNumber = req.GetBankAccountNumber().GetValue()
+	Vendor.Status = "submitted"
+	t := time.Now()
+	Vendor.SubmitTime = &t
+	Vendor.UpdateTime = time.Now()
+	Vendor.StatusTime = time.Now()
```
Comment (chilianyi) [inline pkg/models/vendor_verify_info.go]: Why not use t?

File: pkg/models/vendor_verify_info.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	return &Vendor
+}
+
+func (vendor *AppVendor) ParseVendorSet2pbSet(ctx context.Context, invendors []*AppVendor) []*pb.VendorVerifyInfo {
```
Comment (chilianyi) [inline pkg/models/vendor_verify_info.go]: Change the func name to ParseVendorSet2PbSet

File: pkg/models/vendor_verify_info.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	BankAccountNumber string
+	Status            string
+	RejectMessage     string
+	SubmitTime        *time.Time
+	StatusTime        time.Time
+	UpdateTime        time.Time
+}
+
+func (vendor *AppVendor) ParseInputParams2Obj(req *pb.SubmitVendorVerifyInfoRequest) *AppVendor {
+	Vendor := AppVendor{}
```
Comment (chilianyi) [inline pkg/models/vendor_verify_info.go]: The parameter `Vendor` no need to begin with capitalization.

File: pkg/models/vendor_verify_info.go (hunk 3 above)
Comment (chilianyi) [reply pkg/models/vendor_verify_info.go]: If `invendors` is not one word, change to inVendors will be better.

File: pkg/models/vendor_verify_info.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	var pbVendors []*pb.VendorVerifyInfo
+	for _, invendor := range invendors {
+		var pbVendor *pb.VendorVerifyInfo
+		pbVendor = vendor.ParseVendor2pb(ctx, invendor)
+		pbVendors = append(pbVendors, pbVendor)
+	}
+	return pbVendors
+}
+
+func (vendor *AppVendor) ParseVendor2pb(ctx context.Context, invendor *AppVendor) *pb.VendorVerifyInfo {
```
Comment (chilianyi) [inline pkg/models/vendor_verify_info.go]: Change `ParseVendor2pb` to `ParseVendor2Pb`

File: pkg/plugins/vmbased/metadata_v1.go [hunk 6]
```diff
@@ -0,0 +1,346 @@
+// Copyright 2018 The OpenPitrix Authors. All rights reserved.
```
Comment (chilianyi) [inline pkg/plugins/vmbased/metadata_v1.go]: The file has already been deleted, have you changed any code in the file？

File: pkg/service/appvendor/handler.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
+package appvendor
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/db"
+	"openpitrix.io/openpitrix/pkg/gerr"
+	"openpitrix.io/openpitrix/pkg/manager"
+
```
Comment (chilianyi) [inline pkg/service/appvendor/handler.go]: Useless blank line

File: pkg/service/appvendor/handler.go [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+	"openpitrix.io/openpitrix/pkg/logger"
+	"openpitrix.io/openpitrix/pkg/models"
+	"openpitrix.io/openpitrix/pkg/pb"
+	"openpitrix.io/openpitrix/pkg/util/pbutil"
+)
+
+type Handler struct {
+}
+
+func (h *Handler) DescribeVendorVerifyInfos(ctx context.Context, req *pb.DescribeVendorVerifyInfosRequest) (*pb.DescribeVendorVerifyInfosResponse, error) {
```
Comment (chilianyi) [inline pkg/service/appvendor/handler.go]: Use `func (p *Server) ` here will be better, then no need to add `serverImpl.go`.

File: pkg/models/vendor_verify_info.go (hunk 2 above)
Comment (huojiao2006) [reply pkg/models/vendor_verify_info.go]: There is a requirement to insert one row into  the app vendor table with SubmitTime  as value of Null, so this field should be a point type.

File: pkg/models/vendor_verify_info.go (hunk 1 above)
Comment (huojiao2006) [reply pkg/models/vendor_verify_info.go]: ok

File: pkg/models/vendor_verify_info.go (hunk 3 above)
Comment (huojiao2006) [reply pkg/models/vendor_verify_info.go]: ok

File: pkg/models/vendor_verify_info.go (hunk 5 above)
Comment (huojiao2006) [reply pkg/models/vendor_verify_info.go]: ok

File: pkg/plugins/vmbased/metadata_v1.go (hunk 6 above)
Comment (huojiao2006) [reply pkg/plugins/vmbased/metadata_v1.go]: No,it is a wrong commit.I will remove it.

File: pkg/service/appvendor/handler.go (hunk 7 above)
Comment (huojiao2006) [reply pkg/service/appvendor/handler.go]: ok

**Pull request closed without merging**